### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by gabrielpoca**
This change addresses three issues:

1. Sometimes, when an included file fails to render, we don't see the error overlay. This is common for files such as `_layout.slime` that are included mode than once. For instance, if `_layout.slime` throws an error when it's included by file `a.slime` and succeeds when it's included by file `b.slime`, the error will clear and we never see it.
2. Errors are not propagated, so even if we know that `_layout.slime` is failing, we don't know why.
3. When we rename a file, if there was an error for the file, we don't clear it and we can never solve it because there's no file with the registered name.

To solve this we make a couple of changes:

1. `%SourceFile{}` has a `:dependency_chain` which contains the file names of every file included during the compilation process. This means that each compilation will generate a unique `:dependency_chain`, which makes it possible to identify exactly which file is throwing an error and where the compilation started.
2. We no longer return an empty string when the included file fails. We return the error and raise it again on the file including it. This ensures that everything in that compilation chain will fail, so there are no invisible errors.
3. When a file is removed we clear its errors.

Solves https://github.com/still-ex/still/issues/118

### CODE REVIEW

**Comment by gabrielpoca** [comment]
@frm bump

**Comment by frm** [comment]
> This means that each compilation will generate a unique name

We actually need this. I think the source file should have the compilation chain identifier, it's very useful for debugging.

File: lib/still/preprocessor.ex [hunk 1]
```diff
@@ -131,32 +131,48 @@ defmodule Still.Preprocessor do
...
             remaining_preprocessors: remaining_preprocessors,
-            source_file: file,
-            stacktrace: __STACKTRACE__
-      end
+            source_file: %{
+              source_file
+              | input_file: "#{file.input_file} -> #{source_file.input_file}"
+            },
+            stacktrace: stacktrace
+        end
```
Comment (frm) [inline lib/still/preprocessor.ex:158]: Can we extract this logic to a helper function? The `if/else`

File: lib/still/preprocessor.ex [hunk 2]
```diff
@@ -131,32 +131,48 @@ defmodule Still.Preprocessor do
...
-            message: inspect(e),
+            message: message,
             preprocessor: preprocessor,
             remaining_preprocessors: remaining_preprocessors,
-            source_file: file,
-            stacktrace: __STACKTRACE__
-      end
+            source_file: %{
+              source_file
+              | input_file: "#{file.input_file} -> #{source_file.input_file}"
```
Comment (frm) [inline lib/still/preprocessor.ex:155]: Would you consider adding a `compilation_chain` field instead of this?

**Review by frm — CHANGES_REQUESTED** [review]

File: lib/still/preprocessor.ex (hunk 2 above)
Comment (gabrielpoca) [reply lib/still/preprocessor.ex:155]: I thought about it but it doesn't solve the issue. The `input_file` is our unique identifier, if I don't change we won't be able to distinguish between errors in included files.

Comment (frm) [reply lib/still/preprocessor.ex:155]: Yeah but we can keep the `input_file` field and simply move this to a new one, right? Use that instead.

Comment (gabrielpoca) [reply lib/still/preprocessor.ex:155]: You know what, this doesn't work. I need to rethink this feature.

Comment (gabrielpoca) [reply lib/still/preprocessor.ex:155]: I'll try something else because this approach will not clear the errors properly.

**Comment by gabrielpoca** [comment]
@frm I was wrong and you were right. I updated the code with your idea. Let me know what you think.

**Review by frm — APPROVED** [review]
This feels more sound, nice one 👍

**Pull request merged**